packages/aws-cdk-lib/aws-ecs-patterns/lib/ecs/queue-processing-ecs-service.ts

```diff
@@ -103,9 +103,15 @@ export class QueueProcessingEc2Service extends QueueProcessingServiceBase {
   /**
    * Constructs a new instance of the QueueProcessingEc2Service class.
    */
-  constructor(scope: Construct, id: string, props: QueueProcessingEc2ServiceProps) {
+  constructor(scope: Construct, id: string, props: QueueProcessingEc2ServiceProps = {}) {
     super(scope, id, props);
 
+    // In the queueProcessingFargateService, props.image is optional, which is why image is optional in the queue-processing-service-base.ts.
+    // However, in the queueProcessingEcsService, image is mandatory, so null checks have been introduced.
+    if (props.image == null) {
+      throw new Error('Cannot create a QueueProcessingEc2Service with props.image is undefined');
+    }
```

Comment:
```suggestion
    if (!props.image) {
      throw new Error('image must be specified for EC2 queue processing service');
    }
```
Documentation is already provided in the property's comments.